### PR TITLE
reduce logging and reorder input validation

### PR DESCRIPTION
**Pull request by edoshor**

### CODE REVIEW

File: api/handlers.go [hunk 1]
```diff
@@ -62,6 +62,11 @@ func handleAppend(c *gin.Context, r AppendRequest) (*AppendResponse, *httputil.H
 	if r.ClientEventType == "" {
 		return nil, httputil.NewBadRequestError(errors.New("expected client_event_type to not be empty"))
 	}
+	if r.Data.Valid {
```
Comment (bbfsdev) [inline api/handlers.go:65]: What is not Valid?

File: api/handlers.go [hunk 2]
```diff
@@ -82,17 +88,8 @@ func handleAppend(c *gin.Context, r AppendRequest) (*AppendResponse, *httputil.H
 		entry.UserID = fmt.Sprintf("client:%s", valueOrEmpty(r.ClientId))
 	}
 
-	log := c.MustGet("LOGGER").(zerolog.Logger)
-
-	entry.Data = r.Data
```
Comment (bbfsdev) [inline api/handlers.go:87]: Where this happens on the right side?

Comment (edoshor) [reply api/handlers.go:87]: New line 82

File: api/handlers.go (hunk 1 above)
Comment (edoshor) [reply api/handlers.go:65]: Missing value. i.e. nil

**Pull request merged**